catalyst-gateway/bin/src/db/event/config/key.rs

```diff
@@ -0,0 +1,90 @@
+//! Configuration Key
+
+use std::net::IpAddr;
+
+use serde_json::Value;
+
+/// Configuration key
+#[derive(Debug, Clone, PartialEq)]
+pub(crate) enum ConfigKey {
+    /// Frontend general configuration
+    Frontend,
+    /// Frontend configuration for a specific IP address
+    FrontendForIp(IpAddr),
+}
+
+/// Frontend JSON schema
+const FRONTEND_JSON_SCHEMA: &str = include_str!("jsonschema/frontend.json");
+/// Default frontend configuration
+const FRONTEND_DEFAULT: &str = include_str!("default/frontend.json");
+
+impl ConfigKey {
+    /// Create a `ConfigKey` from ids.
+    pub(crate) fn from_id(id1: &str, id2: &str, id3: &str) -> Option<Self> {
```

Comment:
We should not need this function, because we should not expose "id1", etc outside the database.  All keys should be defined solely through the values in the enum.